From 1e04dcec491bd8f48b5b74ce3e8414132578a645 Mon Sep 17 00:00:00 2001
From: Florian Weimer <fweimer@redhat.com>
Date: Thu, 19 Oct 2023 09:17:38 +0200
Subject: [PATCH 30/39] Revert "elf: Move l_init_called_next to old place of
 l_text_end in link map"

This reverts commit d3ba6c1333b10680ce5900a628108507d9d4b844.

Reason: Preserve internal ABI.
---
 include/link.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/link.h b/include/link.h
index a02d5f2eba..69bda3ed17 100644
--- a/include/link.h
+++ b/include/link.h
@@ -256,10 +256,6 @@ struct link_map
     /* End of the executable part of the mapping.  */
     ElfW(Addr) l_text_end;
 
-    /* Linked list of objects in reverse ELF constructor execution
-       order.  Head of list is stored in _dl_init_called_list.  */
-    struct link_map *l_init_called_next;
-
     /* Default array for 'l_scope'.  */
     struct r_scope_elem *l_scope_mem[4];
     /* Size of array allocated for 'l_scope'.  */
@@ -282,6 +278,10 @@ struct link_map
     /* List of object in order of the init and fini calls.  */
     struct link_map **l_initfini;
 
+    /* Linked list of objects in reverse ELF constructor execution
+       order.  Head of list is stored in _dl_init_called_list.  */
+    struct link_map *l_init_called_next;
+
     /* List of the dependencies introduced through symbol binding.  */
     struct link_map_reldeps
       {
-- 
2.43.0

